mlflow/spark.py

```diff
@@ -660,6 +667,34 @@ def _load_model(model_uri, dfs_tmpdir_base=None):
     return PipelineModel.load(model_uri)
 
 
+def _log_model_via_mlflowdbfs(run_root_artifact_uri, artifact_path, model):
+    artifact_repo = get_artifact_repository(artifact_uri=run_root_artifact_uri)
+    mlflowdbfs_path = artifact_repo._get_mlflowdbfs_path(artifact_path)
+    artifact_repo._set_databricks_host_creds_to_credential_context()
```

Comment:
See https://github.com/mlflow/mlflow/pull/5759/files#r859321751. I don't think these methods should be included in the artifact repo. This is also technically dangerous if the user has set a different artifact root URI for the MLflow experiment, as these methods wouldn't exist. In fact, we shouldn't be using `mlflowdbfs` at all if the artifact root URI isn't of the form `dbfs:/databricks/mlflow-tracking`.